inlong-sort/sort-single-tenant/src/test/java/org/apache/inlong/sort/singletenant/flink/parser/MetaFieldSyncTest.java

```diff
@@ -0,0 +1,184 @@
+import org.apache.inlong.sort.protocol.node.format.JsonFormat;
+import org.apache.inlong.sort.protocol.node.load.KafkaLoadNode;
+import org.apache.inlong.sort.protocol.transformation.FieldRelationShip;
+import org.apache.inlong.sort.protocol.transformation.FilterFunction;
+import org.apache.inlong.sort.protocol.transformation.StringConstantParam;
+import org.apache.inlong.sort.protocol.transformation.function.SingleValueFilterFunction;
+import org.apache.inlong.sort.protocol.transformation.operator.EmptyOperator;
+import org.apache.inlong.sort.protocol.transformation.operator.EqualOperator;
+import org.apache.inlong.sort.protocol.transformation.relation.NodeRelationShip;
+import org.apache.inlong.sort.singletenant.flink.parser.impl.FlinkSqlParser;
+import org.apache.inlong.sort.singletenant.flink.parser.result.FlinkSqlParseResult;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Test for {@link FlinkSqlParser#parseMetaField(Node, BuiltInFieldInfo, StringBuilder)}
+ */
+public class MetaFieldSyncTest extends AbstractTestBase {
+
+    private MySqlExtractNode buildMySQLExtractNode() {
+        List<FieldInfo> fields = Arrays.asList(new FieldInfo("id", new LongFormatInfo()),
+                new FieldInfo("name", new StringFormatInfo()),
+                new FieldInfo("age", new IntFormatInfo()),
+                new FieldInfo("salary", new FloatFormatInfo()),
+                new FieldInfo("ts", new TimestampFormatInfo()),
+                new BuiltInFieldInfo("database", new TimestampFormatInfo(), BuiltInField.MYSQL_METADATA_DATABASE),
+                new BuiltInFieldInfo("table", new TimestampFormatInfo(), BuiltInField.MYSQL_METADATA_TABLE),
+                new BuiltInFieldInfo("pk_names", new TimestampFormatInfo(), BuiltInField.METADATA_PK_NAMES),
+                new BuiltInFieldInfo("event_time", new TimestampFormatInfo(), BuiltInField.MYSQL_METADATA_EVENT_TIME),
+                new BuiltInFieldInfo("event_type", new TimestampFormatInfo(), BuiltInField.MYSQL_METADATA_EVENT_TYPE),
+                new BuiltInFieldInfo("isddl", new TimestampFormatInfo(), BuiltInField.MYSQL_METADATA_IS_DDL),
+                new BuiltInFieldInfo("batch_id", new TimestampFormatInfo(), BuiltInField.METADATA_BATCH_ID),
+                new BuiltInFieldInfo("mysql_type", new TimestampFormatInfo(), BuiltInField.METADATA_MYSQL_TYPE),
+                new BuiltInFieldInfo("sql_type", new TimestampFormatInfo(), BuiltInField.METADATA_SQL_TYPE),
+                new BuiltInFieldInfo("meta_ts", new TimestampFormatInfo(), BuiltInField.METADATA_TS),
+                new BuiltInFieldInfo("up_before", new TimestampFormatInfo(), BuiltInField.METADATA_UPDATE_BEFORE)
+        );
+        Map<String, String> map = new HashMap<>();
+//        map.put("append-mode", "true");
```

Comment:
Does this line necessary?

Reply:
It is not necessary, I remove it latter.